src/lib/rpc/support.ts

```diff
@@ -1,18 +1,11 @@
-import { toastStore } from "$lib/stores/ToastStore";
 import { saveFilePrompt } from "$lib/utils/file";
-import { invoke } from "@tauri-apps/api/tauri";
-import { exceptionLog } from "./logging";
+import { invoke_rpc } from "./rpc";
 
-export async function generateSupportPackage(): Promise<boolean> {
-  try {
-    const savePath = await saveFilePrompt(
-      "ZIP",
-      ["zip"],
-      "opengoal-support-package.zip"
-    );
-    return await invoke("generate_support_package", { userPath: savePath });
-  } catch (e) {
-    exceptionLog("Unable to generate support package", e);
-    toastStore.makeToast("Unable to create support package", "error");
-  }
+export async function generateSupportPackage(): Promise<void> {
+  const userPath = await saveFilePrompt(
+    "ZIP",
+    ["zip"],
+    "opengoal-support-package.zip"
+  );
+  return await invoke_rpc("generate_support_package", { userPath }, () => {});
```

Comment:
it'd be nice to have a variant of `invoke_rpc` that you can provide an explicit toast message if an error occurs -- as it stands all toats will just a stringification of the error and it'd be nice for this to continue to be something more unique/descriptive.  In this case it was `Unable to create support package`.  Then as I mentioned -- those strings can be wired up to be translated.

Reply:
Ah, that sounds neat.

I was considering if there should be a way to avoid showing a toast at all even. Do you think there is a need for that?

Perhaps errors are always logged, but a toast is only shown if the typescript code calling `invoke_rpc` provides an `toastOnError` string?

Reply:
Allright, updated as discussed.

Callers of `invoke_rpc` can now provide a toast message. If provided, it will be displayed on RPC errors, otherwise a default is always shown "An unexpected error occurred".

Side note: I could not find an easy way to integrate svelte-i18n outside of svelte pages, so I left that for now.

The underlying error is logged, see for example (don't mind the icon and the red, I'm just testing locally some ideas for the toasts):
![Skärmavbild 2023-07-07 kl  18 44 01](https://github.com/open-goal/launcher/assets/33322/059612c8-add5-43dd-bd2c-41391321e813)

Reply:
> Side note: I could not find an easy way to integrate svelte-i18n outside of svelte pages, so I left that for now.

For posterity - https://github.com/kaisermann/svelte-i18n/blob/main/docs/FAQ.md#can-i-use-the-formatter-functions-outside-of-a-svelte-component